posts/2018-06-07-shared-state-in-fp.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: Shared State in Functional Programming
+
+meta:
+  nav: blog
+  author: gvolpe
+  pygments: true
+
+tut:
+  scala: 2.12.4
+  binaryScala: "2.12"
+  dependencies:
+    - org.scala-lang:scala-library:2.12.4
+    - org.typelevel::cats-core:1.1.0
+
+---
+
+Newcomers to Functional Programming are often very confused about the proper way to share state without breaking purity and end up having a mix of pure and impure code that defeats the purpose of having pure FP code in the first place.
+
+This is the reason that has motivated me to write a beginner friendly guide :)
+
+## Referential Transparency
+
+I think this is the key to understanding the benefits of Functional Programming. Without it, it'll be hard to understand other deeper concepts, such as `Shared State`. So let's start this guide by refreshing our knowledge about it:
+
+##### What is Referential Transparency?
+
+Being able to substitute an expression by its bound value without changing the meaning of the program.
```

Comment:
The use of "bound value" might suggest this only applies to variable assignment/capture, in general RT says a given expression *e* is referentially transparent if you can evaluate it and substitute its value in **any** larger expression it appears in without changing the meaning of the larger expression.